tests/optimisation_test/recipe_converter.py

```diff
@@ -0,0 +1,61 @@
+import pandas as pd
+import yaml
+from archemist.core.persistence.yaml_handler import YamlHandler
+from nested_lookup import nested_update
+from pathlib import Path
+
+class RecipeUpdate:
+    def __init__(self) -> None:
+        self.placeholder_keys = []
+        self.placeholders =[]
+        super().__init__()
+
+    def find_placeholders(self, data):
+        self._data = data
+        if isinstance(self._data, dict):
+            for key, value in self._data.items():
+                if isinstance(value, str) and '{{' in value and '}}' in value:
+                    self.placeholder_keys.append(key)
+                    self.placeholders.append(value)
+                else:
+                    self.find_placeholders(value)
+        elif isinstance(self._data, list):
+            for item in self._data:
+                self.find_placeholders(item)
+             
+    def update_values(self,data, *args):
```

Comment:
since this method is internal, it would be better if we pass the arguments directly instead of using *args or **kwargs